### PR TITLE
finished answers.md

### PR DESCRIPTION
**Pull request by jjamaltwin**

### CODE REVIEW

**Comment by jjamaltwin** [comment]
@VivaCode

File: Answers.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
 # Answers
 
 1.  What is React JS and what problems does it try and solve?
-
-1.  What does it mean to _think_ in react?
-
-1.  Briefly describe some of the differences between a Class/Stateful component and a Functional/Presentational component.
-
+It can handle complex updates in a fast time. It's modular , so small pieces of code can be reused over and over again. 
+2.  What does it mean to _think_ in react?
```
Comment (VivaCode) [inline Answers.md]: to add on to this, you want to look at each components and assess how many different things its trying to handle. always break it down to its smallest piece.

File: Answers.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-
-1.  What does it mean to _think_ in react?
-
-1.  Briefly describe some of the differences between a Class/Stateful component and a Functional/Presentational component.
-
+It can handle complex updates in a fast time. It's modular , so small pieces of code can be reused over and over again. 
+2.  What does it mean to _think_ in react?
+Think about how you want to structure yuor components, what will be components, and how to reuse them efficiently. 
+3.  Briefly describe some of the differences between a Class/Stateful component and a Functional/Presentational component.
+  A functional component is reg javascript functions that accepts props as an argument and returns a react element.  A class component must extend from React.Component and uses a render function that returns a React element.
```
Comment (VivaCode) [inline Answers.md]: the biggest difference is state. Class components handle, and maintain state. Functional components only receive the necessary props of that state.

File: Answers.md [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 1.  Describe state.
-
```
Comment (VivaCode) [inline Answers.md]: state is the data. it is immuteable.

File: Answers.md [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+    javascript object that is set within the component, used to pass information that outputs in render.
 1.  Describe props.
+    also a javascript object that passes information or functions, but is passed to a component in a function paramerter().
```
Comment (VivaCode) [inline Answers.md]: props is the most recent iteration of state passed down to functional components to be output. It is not always passed as a function - from class it is most often passed as a variable.

**Pull request merged**